chalk-ir/src/lib.rs

```diff
@@ -585,6 +585,49 @@ pub struct Fn<I: Interner> {
     pub substitution: Substitution<I>,
 }
 
+#[derive(Clone, PartialEq, Eq, Hash, PartialOrd, Ord, HasInterner)]
+pub struct Const<I: Interner> {
+    interned: I::InternedConst,
+}
+
+impl<I: Interner> Const<I> {
+    pub fn new(interner: &I, data: impl CastTo<ConstData<I>>) -> Self {
+        Const {
+            interned: I::intern_const(interner, data.cast(interner)),
+        }
+    }
+
+    pub fn interned(&self) -> &I::InternedConst {
+        &self.interned
+    }
+
+    pub fn data(&self, interner: &I) -> &ConstData<I> {
+        I::const_data(interner, &self.interned)
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Hash, HasInterner)]
+pub struct ConstData<I: Interner> {
+    // Should chalk store the type???
+    // I feel like we need it type to deal with 
+    // universally qualified stuff?
```

Comment:
[rustc has it](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.Const.html#structfield.ty), so it seems right in chalk too